test/core/security/tls_security_connector_test.cc

```diff
@@ -737,6 +737,44 @@ TEST_F(TlsSecurityConnectorTest,
                             on_peer_checked);
 }
 
+TEST_F(TlsSecurityConnectorTest,
+       ChannelSecurityConnectorWithVerifiedRootCertSubjectSucceeds) {
+  auto* sync_verifier_ = new SyncExternalVerifier(true);
+  ExternalCertificateVerifier core_external_verifier(sync_verifier_->base());
+  RefCountedPtr<grpc_tls_credentials_options> options =
+      MakeRefCounted<grpc_tls_credentials_options>();
+  options->set_verify_server_cert(true);
+  options->set_certificate_verifier(core_external_verifier.Ref());
+  options->set_check_call_host(false);
+  RefCountedPtr<TlsCredentials> credential =
+      MakeRefCounted<TlsCredentials>(options);
+  ChannelArgs new_args;
+  RefCountedPtr<grpc_channel_security_connector> connector =
+      credential->create_security_connector(nullptr, kTargetName, &new_args);
+  EXPECT_NE(connector, nullptr);
+  TlsChannelSecurityConnector* tls_connector =
+      static_cast<TlsChannelSecurityConnector*>(connector.get());
+  EXPECT_NE(tls_connector->ClientHandshakerFactoryForTesting(), nullptr);
+  // Construct a basic TSI Peer.
+  char* expected_subject = const_cast<char*>(
+      "CN=testca,O=Internet Widgits Pty Ltd,ST=Some-State,C=AU");
+  tsi_peer peer;
+  GPR_ASSERT(tsi_construct_peer(2, &peer) == TSI_OK);
```

Comment:
Can we use EXPECT_EQ here and throughout?

Reply:
We should be able to, but this is another of those patterns taken from the existing tests, so open question of changing just here, or keeping the pattern and doing a fix PR to clean the whole test file another time.

Reply:
I think we should change this now - using GPR_ASSERT is the old style and we should be migrating away from it in new code, IMO.

Same for char* to std::string.

Reply:
Updated the new tests to std::strings and EXPECT_EQ